core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -128,9 +127,9 @@ public long targetSplitSize() {
   }
 
   private static CloseableIterable<ManifestFile> allDataManifestFiles(List<Snapshot> snapshots) {
-    try (CloseableIterable<ManifestFile> iterable = new ParallelIterable<>(
-        Iterables.transform(snapshots, snapshot -> (Iterable<ManifestFile>) () -> snapshot.dataManifests().iterator()),
-        ThreadPools.getWorkerPool())) {
+    try (CloseableIterable<ManifestFile> iterable = CloseableIterable.combine(
+        Iterables.transform(snapshots, snapshot -> CloseableIterable.withNoopClose(snapshot.dataManifests())),
```

Comment:
I don't think that this should add a noop close here. Can you avoid adding this in all the updated calls to create a parallel iterable?

Reply:
I think I can't avoid adding this, becasue `combine` will call `concat` when `workerPool` is null, and `concat` receive `Iterable<CloseableIterable<E>>` but not `Iterable<Iterable<E>>`. So I need to add a noop close to wrap it and pass to `combine`.

Reply:
This does need to be fixed. `ParallelIterable` was constructed using `Iterable<? extends Iterable<T>>`. There should be a version of `CloseableIterable.combine` that accepts the same type. That may mean updating `CloseableIterable.concat` to accept the same `? extends Iterable<T>` in addition to strictly a `CloseableIterable<T>`. But that should be okay, since you can update to close the iterable if it is closeable.